internal/sync/sync.go

```diff
@@ -108,115 +100,44 @@ func repoSynced(clonedDir, localDir string, repo *repo) bool {
 		path.Base(clonedDir), path.Base(localDir))
 
 	err := refactor(
-		fmt.Sprintf("package %s", path.Base(localDir)),
 		fmt.Sprintf("package %s", path.Base(clonedDir)),
-		fmt.Sprintf("./%s", localDir), "*.go")
+		fmt.Sprintf("package %s", path.Base(localDir)),
+		clonedDir, "*.go")
 
 	if err != nil {
-		glog.V(100).Infof("Failed to refactor file before sync due to %w. Exit with error 1", err)
+		glog.V(100).Infof("Failed to replace package names due to %w. Exit with error 1", err)
 		os.Exit(1)
 	}
 
-	glog.V(100).Infof("Replace cloned package imports")
-
 	for _, importMap := range repo.ReplaceImports {
-		err = refactor(
-			importMap["new"],
-			importMap["old"],
-			localDir,
-			"*.go")
-
+		err = refactor(importMap["old"], importMap["new"], clonedDir, "*.go")
 		if err != nil {
-			glog.V(100).Infof("Failed to refactor file. Exit with error 1")
+			glog.V(100).Infof("Failed to refactor files due to %w. Exit with error 1", err)
 			os.Exit(1)
 		}
 	}
-
-	err = execCmd("", "diff", []string{clonedDir, localDir})
-
-	if err == nil {
-		glog.V(100).Infof("Repo synced. Revert local files to original state")
-
-		return true
-	}
-
-	return false
 }
 
-func syncDirectories(clonedDir, localDir string, repo *repo) {
-	glog.V(100).Infof("Repos are not synced. Cleaning local directory: %s", localDir)
-
-	if os.RemoveAll(localDir) != nil {
-		glog.V(100).Infof("Failed to remove local api directory. Exit with error code 1")
-		os.Exit(1)
-	}
-
-	glog.V(100).Infof("Create new local directory: %s", localDir)
+func copyClonedToLocal(clonedDir, localDir string) {
+	glog.V(100).Infof("Create path to new local directory: %s", localDir)
 
 	if os.MkdirAll(localDir, 0750) != nil {
-		glog.V(100).Infof("Failed to recreate api directory. Exit with error code 1")
+		glog.V(100).Info("Failed to create local directory. Exit with error code 1")
 		os.Exit(1)
 	}
 
-	if len(repo.Excludes) > 0 {
-		glog.V(100).Infof("Remove excluded files under %s",
-			path.Base(clonedDir))
-
-		err := excludeFiles(clonedDir, repo.Excludes...)
-		if err != nil {
-			glog.V(100).Infof("Failed to remove excluded files due to %w. Exit with error 1", err)
-			os.Exit(1)
-		}
-	}
-
-	glog.V(100).Infof("Copy api filed from cloned directory to local api directory")
-
-	err := execCmd(
-		"",
-		"cp",
-		[]string{"-a", fmt.Sprintf("%s/.", clonedDir), fmt.Sprintf("%s/", localDir)})
-
+	// We do MkdirAll even though we do RemoveAll so that the path to localDir exists.
+	err := os.RemoveAll(localDir)
```

Comment:
Can you explain why we create the directory and then remove the directory?

Reply:
Not sure I follow based on the comment alone

Reply:
@trewest We create it to make sure the path leading up to it exists and then just delete the specific localDir only. I updated the comments to make this more clear